### PR TITLE
Adding an animation tab

### PR DESCRIPTION
**Pull request by OfficialFreline**
New additional tab "Animations", which allows you to run certain animations on any model.

### CODE REVIEW

**Comment by OfficialFreline** [comment]
Screenshot: https://i.imgur.com/sjzdVq9.jpeg

File: garrysmod/gamemodes/sandbox/gamemode/editor_player.lua [hunk 1]
```diff
@@ -115,6 +115,33 @@ list.Set( "DesktopWindows", "PlayerEditor", {
...
+
+		local animsTable = {
+			"idle_all_scared",
+			"menu_gman",
+			"menu_zombie_01",
+			"menu_combine",
+			"pose_standing_02",
+		}
+
+		for v, n in pairs( animsTable ) do
```
Comment (FlorianLeChat) [inline garrysmod/gamemodes/sandbox/gamemode/editor_player.lua:132]: You can use ipairs here to be more efficient.

Comment (OfficialFreline) [reply garrysmod/gamemodes/sandbox/gamemode/editor_player.lua]: > You can use ipairs here to be more efficient.

Thank you

**Comment by mcNuggets1** [comment]
Very cool and innovative.
But wouldn't there be a way to retrieve all working animations and make this far more extensive?

File: garrysmod/gamemodes/sandbox/gamemode/editor_player.lua (hunk 1 above)
Comment (ExtReMLapin) [reply garrysmod/gamemodes/sandbox/gamemode/editor_player.lua]: You're wrong @FlorianLeChat 
It's not more efficent here, as ipairs is only faster when JIT-compiled, here is not going to be JIT-compiled
On literally each line there is a NYI(closure)/C api call.
When not JIT-compiled, `ipairs` is 3.29 times slower than `pairs`


It is WRONG to use pairs, but not slower.

**Comment by ExtReMLapin** [comment]
Didn't gmod already have in the icon editor an existing sequence lister with preview ?
You could also use `ent:GetSequenceList()` instead of hardcoding the sequences strings.

I see no point of this PR as you can already see the animations in the icon editor.
Playermodel menu is meant to be used by players, there have no need to see this kind of menu, icon editor is meant to be used by "advanced users".

File: garrysmod/gamemodes/sandbox/gamemode/editor_player.lua (hunk 1 above)
Comment (FlorianLeChat) [reply garrysmod/gamemodes/sandbox/gamemode/editor_player.lua]: Good point. I didn't notice that.

**Comment by OfficialFreline** [comment]
> Didn't gmod already have in the icon editor an existing sequence lister with preview ?
> You could also use `ent:GetSequenceList()` instead of hardcoding the sequences strings.
> 
> I see no point of this PR as you can already see the animations in the icon editor.
> Playermodel menu is meant to be used by players, there have no need to see this kind of menu, icon editor is meant to be used by "advanced users".

This addition will allow the same players who make the models to fully finish and test their model. Also, weapon makers wouldn't mind using this either.

**Comment by OfficialFreline** [comment]
Demonstration: https://youtu.be/u_Wb5003Mo0

**Comment by ExtReMLapin** [comment]
why did you move back to `pairs` are you even reading

**Comment by OfficialFreline** [comment]
> why did you move back to `pairs` are you even reading

Small banter) The world is not without fun

**Comment by OfficialFreline** [comment]
And yes, I spoke to some content creators today. The idea of ​​the tab was appreciated

**Comment by ExtReMLapin** [comment]
already exists for fuck sake

**Comment by OfficialFreline** [comment]
> 
> already exists for fuck sake

This is a little different

**Pull request closed without merging**